packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,497 @@
+  void updateRenderObject(BuildContext context, _RenderSegmentedControl renderObject) {
+    renderObject.markNeedsPaint();
+    renderObject
+      ..children = children
+      ..selectedValue = selectedValue
+      ..pressedValue = pressedValue;
+  }
+}
+
+class _RenderSegmentedControlElement extends RenderObjectElement {
+  _RenderSegmentedControlElement(_SegmentedControlRenderWidget segmentedControl)
+      : super(segmentedControl);
+
+  final List<Element> slots = List<Element>();
+
+  @override
+  _SegmentedControlRenderWidget get widget => super.widget;
+
+  @override
+  _RenderSegmentedControl get renderObject => super.renderObject;
+
+  @override
+  void update(_SegmentedControlRenderWidget newWidget) {
+    super.update(newWidget);
+    assert(widget == newWidget);
+    for(int i = 0; i < widget.children.length; i += 1) {
+      _updateChild(widget.children.elementAt(i), i);
+    }
+  }
+
+  void _updateChild(Widget widget, int slot) {
+    final Element oldChild = slots.elementAt(slot);
+    final Element newChild = updateChild(oldChild, widget, slot);
+    if(oldChild != null) {
+      slots.remove(oldChild);
+    }
+    if(newChild != null) {
+      slots.insert(slot, newChild);
+    }
+  }
+
+  @override
+  void visitChildren(ElementVisitor visitor) {
+    slots.forEach(visitor);
+  }
+
+  @override
+  void forgetChild(Element child) {
+    assert(slots.contains(child));
+    slots.remove(child);
+  }
+
+  @override
+  void insertChildRenderObject(RenderObject child, dynamic value) {
+    assert(child is RenderBox);
+    assert(value is int);
+    assert((value as int) <= slots.length);
+    renderObject.setBox(value, child);
+    assert(renderObject.slots.contains(child));
+  }
+
+  @override
+  void removeChildRenderObject(RenderObject child) {
+    assert(child is RenderBox);
+    assert(renderObject.slots.contains(child));
+    renderObject.setBox(renderObject.boxes.indexOf(child), null);
+    assert(!renderObject.slots.contains(child));
+  }
+
+  @override
+  void moveChildRenderObject(RenderObject child, dynamic value) {
+    assert(false, 'not reachable');
+  }
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    for (int i = 0; i < widget.children.length; i += 1) {
+      _mountChild(widget.children.elementAt(i), i);
+    }
+  }
+
+  void _mountChild(Widget widget, int slot) {
+    Element oldChild;
+    if (slot >= slots.length) {
+      oldChild = null;
+    } else {
+      oldChild = slots.elementAt(slot);
+    }
+    final Element newChild = updateChild(oldChild, widget, slot);
+    if (oldChild != null) {
+      slots.remove(oldChild);
+    }
+    if (newChild != null) {
+      slots.insert(slot, newChild);
+    }
+  }
+}
+
+class _RenderSegmentedControl extends RenderBox {
+  _RenderSegmentedControl(
+      {@required this.children, @required this.selectedValue, @required this.pressedValue});
+
+  List<Widget> children;
+
+  Key selectedValue, pressedValue;
```

Comment:
Put these on separate lines instead of using a comma. Also, I think they can be final.